platform-sdk/docs/hashgraph_glossary.md

```diff
@@ -0,0 +1,250 @@
+  ensures that any given event can see at most one of `x` and `y`, if they form a branch. And for an honest node, if it
+  creates an event that sees one of `x` or `y`, all of its self-descendents will continue to see that same `x` or `y`,
+  and none of them will ever see the other one. The 3 requirements for them to constitute a branch are:
+    - both are created by the same creator
+    - neither is a self-ancestor of the other
+    - the difference of the birth rounds is at most `roundsAncient` (for the value of `roundsAncient` defined in the
+      roster whose roster round equals the max of their birth rounds)
+
+  [top](#top)
+
+<a name="generations"/>
+
+## Generations
+
+- __Local__ generation - an event is assigned this immutable number when it is added to the hashgraph. It is 1 plus the
+  max of all the local generations of its non-ancient parents. If it has no non-ancient parents, then it has a local
+  generation of 1. For a given event, this number can be different on different nodes. It can even be different on a
+  single node, before and after a restart or reconnect. The purpose of this number is to make it easy to sort a set of
+  events into topological order. (A topological order is any ordering of the events such that an event's parents always
+  precede it in the order).
+- __Consensus__ generation - an event is assigned this immutable number when it reaches consensus. It is 1 plus the max
+  of all the consensus generations of its eligible parents, where where an event is eligible if it is not ancient and
+  has not reached consensus in a previous round. If an event has no eligible parents, then its consensus generation is
+  1. This is deterministic, so an event is guaranteed to have the same consensus generation on all nodes. This is used
+  while finding the consensus order, since one step in the process is to break ties by consensus generation, thus
+  ensuring the final order is a topological order. This is easily calculated during the calculation of which events have
+  reached consensus in a given round. The latter is done by doing a depth-first search from each judge. It is easy to
+  add a line of code to this search to do the calculation of consensus generation at the same time, with no additional
+  searching at all.
+
+  [top](#top)
+
+<a name="addressbooks"/>
+
+## Address Books
+
+- __Address book__ - the current info for all nodes, stored as a file in the Hedera file system, and therefore also in
+  state. This includes public keys, IP addresses, node ID, proxies, consensus weight, and other info.
+- __Roster__ - a subset of the address book, with just the information needed for consensus. This includes the nodeID,
+  public key for signing events, consensus weight, number of consensus shares, and some cryptographic info related to
+  TSS and state proofs. Each roster is associated with an unsigned long `roster round`. The roster with a given
```

Comment:
`cryptographic info related to TSS and state proofs` is no longer in the roster.